src/auth-service/bin/jobs/preferences-log-job.js

```
@@ -13,25 +14,41 @@ const logUserPreferences = async () => {
         break;
       }
 
-      // Fetch existing preferences for users in batch
-      const userIds = users.map((user) => user._id);
+      // Filter users who are members of the default group
+      const validUsers = users.filter(
+        (user) =>
+          user.group_roles &&
+          user.group_roles.some(
+            (role) => role.group.toString() === defaultGroupId.toString()
+          )
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)